client/src/Validator/RouteValidator.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Validator;
+
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\RouterInterface;
+
+class RouteValidator
+{
+    public static function validateRoute(RouterInterface $router, string $path): bool
+    {
+        try {
+            $router->match($path);
```

Comment:
Worth adding a few unit tests around this helper to make sure a full url rather than a path doesn't do anything weird, and to cover us for symfony changing behaviour at some future date. 

Basically make sure we avoid any nasty [open redirect](https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html ) issues.

Reply:
You're right, had skipped the tests as there was a lot more work for the other files I had touched. So we've got a ticket to adjust the testing for those. I was being overly trusting of Symfony, will add some now.